GitUI/UserControls/FileStatusWithDescription.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+using GitCommands;
+using GitUIPluginInterfaces;
+
+namespace GitUI
+{
+    public class FileStatusWithDescription
+    {
+        public GitRevision FirstRev;
+        public GitRevision SecondRev;
+        public ObjectId BaseA;
+        public ObjectId BaseB;
+        public string Summary;
+        public IReadOnlyList<GitItemStatus> Statuses;
+    }
```

Comment:
The only thing I would prefer is to turn this into an immutable class, i.e.
```suggestion
    public class FileStatusWithDescription
    {
        public FileStatusWithDescription(GitRevision firstRev, GitRevision secondRev, ObjectId baseA, ObjectId baseB, string summary, IReadOnlyList<GitItemStatus> statuses)
        {
            FirstRev = firstRev ?? throw new ArgumentNullException(nameof(firstRev));
            SecondRev = secondRev ?? throw new ArgumentNullException(nameof(secondRev));
            BaseA = baseA ?? throw new ArgumentNullException(nameof(baseA));
            BaseB = baseB ?? throw new ArgumentNullException(nameof(baseB));
            Summary = summary ?? throw new ArgumentNullException(nameof(summary));
            Statuses = new List<GitItemStatus>();
            if (statuses != null)
            {
                Statuses.AddRange(statuses);
            }
        }
        
        public GitRevision FirstRev { get; }
        public GitRevision SecondRev { get; }
        public ObjectId BaseA { get; }
        public ObjectId BaseB { get; }
        public string Summary { get; }
        public IReadOnlyList<GitItemStatus> Statuses { get; }
    }
```

Reply:
Pushed an update, some changes had to be done

> I have no more comments despite immutable FileStatusWithDescription with [CanBeNull] baseA/B will be good.

As well as firstRev for initial (root) commit.

OT: I found later that the initial commit uses the sha e83c5163316f89bfbde7d9ab23ca2e25604af290 within the Git code base, so [CanBeNull] could be changed...
But not here.